### ISSUE TITLE
Handling images/attachments?

### ISSUE THREAD

**Issue by sanscontext** [justmiles/go-markdown2confluence#66]
Hello again, I'm running into this and I'm not sure if I'm doing something wrong or if this just isn't included.

I have a directory of markdown files exported from a (*cough*) popular wiki like tool. The export structure goes something like this:

```ascii
- Home page
  - page1.md
  - page1 (folder)
     - image1.png
     - image2.png
     - childpage.md
     - childpage (folder)
       - childpageimage.png
   - page2.md
   [...] 
```

Is there a way to make sure these files get attached to the resulting Confluence page?

(If no, then further questions are moot :) )

**Comment by justmiles** [comment]
As it's currently written it'll grab any images referenced. So if you use the markdown `![image](image1.png)` syntax it should pick it up.

It won't, however, just include all nearby images as attachments.

**Comment by sanscontext** [comment]
Can you tell me the expected file structure where it'll grab the images? I'm happy to push files around to the right place if it'll make this work.

**Comment by justmiles** [comment]
It's relevant to the document. If you include an image [using the standard markdown spec](https://www.markdownguide.org/basic-syntax/#images-1) `markdown2confluence` will test to see if that file exists locally. If it does, it will upload it.

So, technically, it can be anywhere on disk it just needs to be included in the markdown itself.

**Comment by justmiles** [comment]
I think this, as well as #14, are going to require some sort of state management to solve for. With the idea being that as we're creating documents in Confluence we're storing the ID of the page created somewhere. We can then translate relative links to confluence pages by referencing this key/value store. 

Thinking through this at the time of sync we'd need to create all the documents and store the IDs, then re-update the ones with nested links. So something like this:

- Upload all documents as it currently does
- On upload, store remote confluence page ID key/value store
- On upload, if a link resolves to relative markdown file, defer to a final fix-links-and-upload step
- Reprocess all deferred files that have links to relative markdown files
- To satisfy #14, delete any Confluence pages that exist in the store, but not on disk.

I haven't started writing it, but am thinking of a simple interface that interacts with a backend key/value store. Maybe start with boltdb?

**Comment by justmiles** [comment]
Yes, the whole thing could be done in one run without persisting storage but not for all use cases. It makes sense for one-time migrations where you're syncing the whole tree, but less so when you're syncing partial files often. I personally use it to sync modified files every 30 minutes.

I'm sure we could make local storage optional, though. If the first pass doesn't have it, it's still worth doing - definitely encourage you to jump in!

**Comment by justmiles** [comment]
This is available in [v4.0.0-alpha-2](https://github.com/justmiles/go-markdown2confluence/releases/tag/v4.0.0-alpha-2)
